caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,235 @@
+        return SequenceClassification._process_predictions(scores_dict, text_idx=0)
+
+    def run_batch(self, texts: List[str]) -> List[ClassificationResult]:
+        """Run the sequence classification on batch, truncates sequences too long for model
+
+        Args:
+            text: List[str]
+                Input strings to be classified
+
+        Returns:
+            List[ClassificationResult]
+        """
+        scores_dict = self._get_scores(texts)
+        num_texts = len(texts)
+
+        # Re-organize scores_dict for each example
+        # We could eventually consider whether or not to sort classifications by scores
+        # but avoiding this prescription here for now
+        classification_predictions = []
+        for text_idx in range(num_texts):
+            classification_prediction = SequenceClassification._process_predictions(
+                scores_dict, text_idx
+            )
+            classification_predictions.append(classification_prediction)
+        return classification_predictions
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        self.tokenizer.save_pretrained(model_path)
+        self.model.save_pretrained(model_path)
+
+    @classmethod
+    def load(cls, model_path: str) -> "SequenceClassification":
+        """Load a tokenizer and sequence classification model. Assumes the
+        tokenizer and model are HuggingFace transformer-based and on
+        the same model_path
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            SequenceClassification
+                Instance of this class built from the on disk model.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(model_path)
+        model = AutoModelForSequenceClassification.from_pretrained(model_path)
+        device = SequenceClassification._get_device()
+        return cls(
+            tokenizer=tokenizer,
+            model=model,
+            device=device,
+        )
+
+    ################################## Private Functions #########################################
+
+    def _get_scores(self, text: Union[str, List[str]]):
+        """Run tokenizer and model to get scores on text(s)
+
+        Args:
+            text: Union[str, List[str]]
+                Input string(s) to be used
+
+        Returns:
+            scores_dict
+                Dict with key label, and values as the array of scores,
+                each corresponding to text(s)
+        """
+        # Apply tokenizer
+        tokenized_text = self.tokenizer(
+            text,
+            padding="max_length",
+            max_length=self.tokenizer.model_max_length,
+            truncation=True,
+            return_tensors="pt",  # PyTorch
+        )
+        # NOTE: no simple/efficient way to detect whether truncation
```

Comment:
The original idea was to notify the user if their input was truncated (in the case that input is longer than the "max length" supportable by the model used). I had left this comment here to indicate that through the transformers API figuring out whether or not the truncation was done was not easily doable. Curious if that still feels misplaced?

Reply:
gotcha. I felt it was a misplaced since I was looking at the code line below this comment which was talking about the "device". 

But also, can we detect truncation by checking if the character at `max_length` is padding token or not ? 
1. if its padding_token, then no truncation happened
2. if its something else, then 

Or actually, `verbose` flag in tokenizer will automatically print the warning.

Or may be we can use the comparison between `return_length` and `tokenizer.max_length` to detect truncation?

Reply:
`return_length` will always be `max_length` when padding is used. 

`verbose` is `True` by default 🤔 on tokenizer calls - https://huggingface.co/docs/transformers/main_classes/tokenizer.  I did not see any warning messages testing this

I did consider the "check particular tokens" thing but wasn't sure if that would be extra overhead since if it's not a padding token it could either be there's exactly `max_length` tokens (so truncation didn't happen) or truncation did happen